mentorship ios/Service/Networking/RelationAPI.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  RelationAPI.swift
+//  Created on 23/07/20
+//  Created for AnitaB.org Mentorship-iOS 
+//
+
+import Foundation
+import Combine
+
+class RelationAPI {
+    private var cancellable: AnyCancellable?
+    private var tasksCancellable: AnyCancellable?
+    
+    func fetchCurrentRelation(completion: @escaping (RequestStructure) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //api call
+        cancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.currentRelation, token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(RelationModel().currentRelation) }
+            .sink {
+                completion($0)
+        }
+    }
+    
+    func fetchTasks(id: Int, completion: @escaping ([TaskStructure], Bool) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        // make api call
+        tasksCancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.getCurrentTasks(id: id), token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(RelationModel().tasks) }
+            .sink {
+                var success = false
+                if NetworkManager.responseCode == 200 {
+                    success = true
+                }
+                completion($0, success)
+        }
+    }
+    
+    //create newtask api call
+    func addNewTask(newTask: RelationModel.AddTaskData, relationID: Int, completion: @escaping (RelationModel.ResponseData) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //prepare upload data
+        guard let uploadData = try? JSONEncoder().encode(newTask) else {
+            return
```

Comment:
Shell we return here a closure with fail?

Reply:
How do you feel about using a fatal error here instead?

Reply:
would UI be able to catch it?